Compiler error in version 2.0.0 with GCC7 - ambiguous use of enum value

Due to stricter rules, `beta` is considered ambiguous and should be `NEST::beta` instead.
It would be even better to define it as `enum class INTERACTION_TYPE`, i.e. `beta` &rarr; `NEST::INTERACTION_TYPE::beta` which is much clearer.
```c++
/code/sim/NEST_LZ/testNEST.cpp: In function ‘int main(int, char**)’:
/code/sim/NEST_LZ/testNEST.cpp:160:16: error: reference to ‘beta’ is ambiguous
     type_num = beta;  // default electron recoil model
                ^~~~
In file included from /code/sim/NEST_LZ/testNEST.cpp:14:0:
/code/sim/NEST_LZ/include/NEST.hh:112:3: note: candidates are: NEST::INTERACTION_TYPE beta
   beta = 8,
   ^~~~
In file included from /cvmfs/sft.cern.ch/lcg/releases/gcc/7.3.0-90605/x86_64-centos7-gcc62-opt/include/c++/7.3.0/cmath:1914:0,
                 from /cvmfs/sft.cern.ch/lcg/releases/gcc/7.3.0-90605/x86_64-centos7-gcc62-opt/include/c++/7.3.0/math.h:36,
                 from /code/sim/NEST_LZ/include/RandomGen.hh:9,
                 from /code/sim/NEST_LZ/include/NEST.hh:65,
                 from /code/sim/NEST_LZ/testNEST.cpp:14:
/cvmfs/sft.cern.ch/lcg/releases/gcc/7.3.0-90605/x86_64-centos7-gcc62-opt/include/c++/7.3.0/bits/specfun.h:343:5: note:                 template<class _Tpa, class _Tpb> typename __gnu_cxx::__promote_2<_Tp, _Up>::__type std::beta(_Tpa, _Tpb)
     beta(_Tpa __a, _Tpb __b)
     ^~~~
/code/sim/NEST_LZ/testNEST.cpp:238:31: error: reference to ‘beta’ is ambiguous
       yieldsMax = n.GetYields(beta, energyMaximum, rho, centralField,
                               ^~~~
In file included from /code/sim/NEST_LZ/testNEST.cpp:14:0:
/code/sim/NEST_LZ/include/NEST.hh:112:3: note: candidates are: NEST::INTERACTION_TYPE beta
   beta = 8,
   ^~~~
In file included from /cvmfs/sft.cern.ch/lcg/releases/gcc/7.3.0-90605/x86_64-centos7-gcc62-opt/include/c++/7.3.0/cmath:1914:0,
                 from /cvmfs/sft.cern.ch/lcg/releases/gcc/7.3.0-90605/x86_64-centos7-gcc62-opt/include/c++/7.3.0/math.h:36,
                 from /code/sim/NEST_LZ/include/RandomGen.hh:9,
                 from /code/sim/NEST_LZ/include/NEST.hh:65,
                 from /code/sim/NEST_LZ/testNEST.cpp:14:
/cvmfs/sft.cern.ch/lcg/releases/gcc/7.3.0-90605/x86_64-centos7-gcc62-opt/include/c++/7.3.0/bits/specfun.h:343:5: note:                 template<class _Tpa, class _Tpb> typename __gnu_cxx::__promote_2<_Tp, _Up>::__type std::beta(_Tpa, _Tpb)
     beta(_Tpa __a, _Tpb __b)
     ^~~~
/code/sim/NEST_LZ/testNEST.cpp:468:30: error: reference to ‘beta’ is ambiguous
         yields = n.GetYields(beta, refEnergy, rho, detector->FitEF(xx, yy, zz),
                              ^~~~
In file included from /code/sim/NEST_LZ/testNEST.cpp:14:0:
/code/sim/NEST_LZ/include/NEST.hh:112:3: note: candidates are: NEST::INTERACTION_TYPE beta
   beta = 8,
   ^~~~
In file included from /cvmfs/sft.cern.ch/lcg/releases/gcc/7.3.0-90605/x86_64-centos7-gcc62-opt/include/c++/7.3.0/cmath:1914:0,
                 from /cvmfs/sft.cern.ch/lcg/releases/gcc/7.3.0-90605/x86_64-centos7-gcc62-opt/include/c++/7.3.0/math.h:36,
                 from /code/sim/NEST_LZ/include/RandomGen.hh:9,
                 from /code/sim/NEST_LZ/include/NEST.hh:65,
                 from /code/sim/NEST_LZ/testNEST.cpp:14:
/cvmfs/sft.cern.ch/lcg/releases/gcc/7.3.0-90605/x86_64-centos7-gcc62-opt/include/c++/7.3.0/bits/specfun.h:343:5: note:                 template<class _Tpa, class _Tpb> typename __gnu_cxx::__promote_2<_Tp, _Up>::__type std::beta(_Tpa, _Tpb)
     beta(_Tpa __a, _Tpb __b)
     ^~~~
/code/sim/NEST_LZ/testNEST.cpp: In function ‘int main(int, char**)’:
/code/sim/NEST_LZ/testNEST.cpp:160:16: error: reference to ‘beta’ is ambiguous
     type_num = beta;  // default electron recoil model
                ^~~~
In file included from /code/sim/NEST_LZ/testNEST.cpp:14:0:
/code/sim/NEST_LZ/include/NEST.hh:112:3: note: candidates are: NEST::INTERACTION_TYPE beta
   beta = 8,
   ^~~~
In file included from /cvmfs/sft.cern.ch/lcg/releases/gcc/7.3.0-90605/x86_64-centos7-gcc62-opt/include/c++/7.3.0/cmath:1914:0,
                 from /cvmfs/sft.cern.ch/lcg/releases/gcc/7.3.0-90605/x86_64-centos7-gcc62-opt/include/c++/7.3.0/math.h:36,
                 from /code/sim/NEST_LZ/include/RandomGen.hh:9,
                 from /code/sim/NEST_LZ/include/NEST.hh:65,
                 from /code/sim/NEST_LZ/testNEST.cpp:14:
/cvmfs/sft.cern.ch/lcg/releases/gcc/7.3.0-90605/x86_64-centos7-gcc62-opt/include/c++/7.3.0/bits/specfun.h:343:5: note:                 template<class _Tpa, class _Tpb> typename __gnu_cxx::__promote_2<_Tp, _Up>::__type std::beta(_Tpa, _Tpb)
     beta(_Tpa __a, _Tpb __b)
     ^~~~
/code/sim/NEST_LZ/testNEST.cpp:238:31: error: reference to ‘beta’ is ambiguous
       yieldsMax = n.GetYields(beta, energyMaximum, rho, centralField,
                               ^~~~
In file included from /code/sim/NEST_LZ/testNEST.cpp:14:0:
/code/sim/NEST_LZ/include/NEST.hh:112:3: note: candidates are: NEST::INTERACTION_TYPE beta
   beta = 8,
   ^~~~
In file included from /cvmfs/sft.cern.ch/lcg/releases/gcc/7.3.0-90605/x86_64-centos7-gcc62-opt/include/c++/7.3.0/cmath:1914:0,
                 from /cvmfs/sft.cern.ch/lcg/releases/gcc/7.3.0-90605/x86_64-centos7-gcc62-opt/include/c++/7.3.0/math.h:36,
                 from /code/sim/NEST_LZ/include/RandomGen.hh:9,
                 from /code/sim/NEST_LZ/include/NEST.hh:65,
                 from /code/sim/NEST_LZ/testNEST.cpp:14:
/cvmfs/sft.cern.ch/lcg/releases/gcc/7.3.0-90605/x86_64-centos7-gcc62-opt/include/c++/7.3.0/bits/specfun.h:343:5: note:                 template<class _Tpa, class _Tpb> typename __gnu_cxx::__promote_2<_Tp, _Up>::__type std::beta(_Tpa, _Tpb)
     beta(_Tpa __a, _Tpb __b)
     ^~~~
/code/sim/NEST_LZ/testNEST.cpp:468:30: error: reference to ‘beta’ is ambiguous
         yields = n.GetYields(beta, refEnergy, rho, detector->FitEF(xx, yy, zz),
                              ^~~~
In file included from /code/sim/NEST_LZ/testNEST.cpp:14:0:
/code/sim/NEST_LZ/include/NEST.hh:112:3: note: candidates are: NEST::INTERACTION_TYPE beta
   beta = 8,
   ^~~~
In file included from /cvmfs/sft.cern.ch/lcg/releases/gcc/7.3.0-90605/x86_64-centos7-gcc62-opt/include/c++/7.3.0/cmath:1914:0,
                 from /cvmfs/sft.cern.ch/lcg/releases/gcc/7.3.0-90605/x86_64-centos7-gcc62-opt/include/c++/7.3.0/math.h:36,
                 from /code/sim/NEST_LZ/include/RandomGen.hh:9,
                 from /code/sim/NEST_LZ/include/NEST.hh:65,
                 from /code/sim/NEST_LZ/testNEST.cpp:14:
/cvmfs/sft.cern.ch/lcg/releases/gcc/7.3.0-90605/x86_64-centos7-gcc62-opt/include/c++/7.3.0/bits/specfun.h:343:5: note:                 template<class _Tpa, class _Tpb> typename __gnu_cxx::__promote_2<_Tp, _Up>::__type std::beta(_Tpa, _Tpb)
     beta(_Tpa __a, _Tpb __b)
     ^~~~
make[2]: *** [CMakeFiles/NEST_core.dir/testNEST.cpp.o] Error 1
make[1]: *** [CMakeFiles/NEST_core.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/NEST.dir/testNEST.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/NEST.dir/all] Error 2
make: *** [all] Error 2
```

--- a/testNEST.cpp
+++ b/testNEST.cpp
@@ -156,7 +156,7 @@ int main(int argc, char** argv) {
            type == "compton" || type == "electron" || type == "e-" ||
            type == "muon" || type == "MIP" || type == "LIP" || type == "mu" ||
            type == "mu-")
-    type_num = beta;  // default electron recoil model
+    type_num = NEST::beta;  // default electron recoil model
   else {
     cerr << "UNRECOGNIZED PARTICLE TYPE!! VALID OPTIONS ARE:" << endl;
     cerr << "NR or neutron," << endl;
@@ -234,7 +234,7 @@ int main(int argc, char** argv) {
     else
       energyMaximum = eMax;
     if (type_num == Kr83m)
-      yieldsMax = n.GetYields(beta, energyMaximum, rho, centralField,
+      yieldsMax = n.GetYields(NEST::beta, energyMaximum, rho, centralField,
                               double(massNum), double(atomNum),
                               NuisParam);  // the reason for this: don't do the
                                            // special Kr stuff when just
@@ -464,7 +464,7 @@ int main(int argc, char** argv) {
              sqrt(pow(xx, 2.) + pow(yy, 2.)) <
                  detector->get_radmax()) {  // stop making S1 and S2 if particle
                                             // exits Xe vol
-        yields = n.GetYields(beta, refEnergy, rho, detector->FitEF(xx, yy, zz),
+        yields = n.GetYields(NEST::beta, refEnergy, rho, detector->FitEF(xx, yy, zz),
                              double(massNum), double(atomNum), NuisParam);
         quanta = n.GetQuanta(yields, rho);
         Nph += quanta.photons * (eStep / refEnergy);

